epl_trends/epl.html

```diff
@@ -8,6 +8,14 @@
     <!-- <script src="https://d3js.org/d3-queue.v3.min.js"></script> -->
     <script src="process_json_data.js"></script>
     <style>
+      
+    .xgrid line,
+    .xgrid path,
+    .ygrid line,
+    .ygrid path {
+      stroke: #ddd;
+    }
```

Comment:
@mwernimont I know there is a much better way to do this ... they are all under a group element with the id `grid`, but I tried just `#grid` and also `#grid path, #grid line` but couldn't get it to work.

Reply:
@lindsaycarr so comparatively this is how I set up my gridlines.

```
chart.append('g')
		.classed('gridlines', true)
		.call(makeYGridlines()
			.ticks(12)
			.tickSize(-chartWidth)
			.tickFormat("")
		)
		.attr('transform', 'translate(0,0)');
```

Using classed allows you to give an element multiple shareable class names for CSS purposes.  So maybe thats a better solution? Or just could be preference on my part.

and my CSS is 

```
.gridlines path,
.gridlines line{
	stroke:rgb(190,190,190);
}

```

Reply:
Keep in mind I'm only doing y gridlines, but maybe classed is worth a shot.

Reply:
SWEET. It worked :) commit on its way!

Reply:
cool! so what's the trick, exactly? is it that you can't do `#id type` but you can do `.class type` in CSS selectors?

Reply:
So I think you can't do a group CSS selection and effect the lines and paths inside the group just by calling that group ID. You could do 

```
#grid.gridlines line,
#grid.gridlines path{
     stroke: #000;
}
```

I'm pretty sure.  Just might be easier and cleaner to read by giving the specific parts you are adding the stroke to, a shared class name.

Reply:
And in my example, I have the shared class name, so no real reason to have the ID in front of it, but you get what I mean.

Reply:
ok, so `#grid path, #grid line` doesn't work, while `#grid.gridlines line, #grid.gridlines path` and `.gridlines line, .gridlines path` do...i'm not getting the intuition, but I suppose I could memorize that. or is there more about that distinction that you can clarify?

Reply:
So #grid has no paths or lines associated with it, so that is why 

```
#grid line,
#grid path{
   stroke:#000;
}
```
doesn't work.  Its just the holder of the elements that can take a stroke.

Reply:
Certain CSS doesn't cascade into children of a parent unless told to.

Reply:
Does that make more sense?

Reply:
ohhh, hmm, maybe! so I'm trying to get a handle on the DOM by reading the code. I think we have:
```
<g id='grid' class='gridlines'>
  <g class='xgrid'>
    <path ...>
    <line ...>
    <...etc...>
  </g>
</g>
```
is that right?
and if so, the is it correct to say this?: While `.gridlines path` works because `.gridlines` is the grandparent of many `path` elements, CSS interprets `#grid path` more strictly to mean a `path` directly within an element with id `#grid`

Reply:
Yes, I think you are on to something.  CSS only goes one selector down after a element selection.  So #grid group goes to .gridlines group and checks for lines or paths, but it doesn't go inside .gridlines group to check for lines or paths.

Reply:
Nice! That's logic I can start to understand and remember. I don't suppose you can find a link to a doc that describes this in any more detail? Something like element vs group selections?

Reply:
https://www.smashingmagazine.com/2007/07/css-specificity-things-you-should-know/

Article from 2007 from Smashing Magazine (CSS, Web Magazine I love) which was updated in 2016 about CSS specificity.

Reply:
sweet, thanks!